src/EscapeAnalysis.jl

```diff
@@ -1969,6 +1982,22 @@ function array_isassigned_nothrow(args::Vector{Any}, src::IRCode)
     return true
 end
 
+is_finalizer(name::Symbol) = name === :jl_gc_add_finalizer_th
+
+function escape_finalizer!(astate::AnalysisState, pc::Int, args::Vector{Any})
+    length(args) ≥ 6 || return add_fallback_changes!(astate, pc, args)
+    println("Finalizer escape!")
+    println(pc)
+    for arg in args
+        println(arg)
+    end
+    value = args[8]
+    func = args[9]
+    # TODO: We need to analyze func
+    add_escape_change!(astate, value, FinalizerEscape(pc))
+    add_liveness_changes!(astate, pc, args, 6)
+end
```

Comment:
```suggestion
function escape_finalizer!(astate::AnalysisState, pc::Int, args::Vector{Any})
    length(args) ≥ 8 || return add_fallback_changes!(astate, pc, args)
    obj = args[7]
    func = args[8]
    # TODO: We need to analyze func
    add_escape_change!(astate, obj, FinalizerEscape(pc))
    add_liveness_changes!(astate, pc, args, 6)
end
```

Reply:
Thanks! I realized this late last night but didn't get around to pushing the changes :smile: